src/main/java/org/jabref/gui/duplicationFinder/DuplicateSearch.java

```diff
@@ -162,10 +162,8 @@ private void handleDuplicates(DuplicateSearchResult result) {
         final NamedCompound compoundEdit = new NamedCompound(Localization.lang("duplicate removal"));
         // Now, do the actual removal:
         if (!result.getToRemove().isEmpty()) {
-            for (BibEntry entry : result.getToRemove()) {
-                panel.getDatabase().removeEntry(entry);
-                compoundEdit.addEdit(new UndoableRemoveEntry(panel.getDatabase(), entry));
-            }
+            compoundEdit.addEdit(new UndoableRemoveEntries(panel.getDatabase(), new ArrayList<>(result.getToRemove())));
```

Comment:
I use the constructor of `UndoableRemoveEntries` accepting a single BibEntry, and similarly the method `BibDatabase#removeEntry`. 
(Btw, there is `Colections.singletonList(element)` which is preferred over `new ArrayList<>(element)`).

Reply:
The problem is that `DuplicateSearch.getToRemove` returns a Collection<BibEntry>, not a single one. The only way I found that I could coerce a `Collection` to a `List` was via the `ArrayList<>` constructor. I could make a constructor for a `Collection`, but that feels to specific.

Reply:
+1 for the second constructor. Even if very specific, the code with `new ArrayList<>` reads strange.